demos/ota/ota_demo_core_http/ota_demo_core_http.c

```diff
@@ -247,7 +247,7 @@
 /**
  * @brief The common prefix for all OTA topics.
  */
-#define OTA_TOPIC_PREFIX                 "$aws/things/"
+#define OTA_TOPIC_PREFIX                 "$aws/things/" CLIENT_IDENTIFIER "/"
```

Comment:
Do we need to put thing name as part of the prefix here. Some devices could be reading thing name from flash then the preprocessing fails. Could this be replaced with a wildcard (as thing name is always constant for this device). Ex: `$aws/things/+/jobs/$next/get/accepted. 
This way the topic filters can be defined as string constants and demo functions needs to call MQTT_MatchTopic() to get the right filter.

Reply:
Update for both OTA MQTT and OTA HTTP demos